asmcomp/asm_target/asm_section.ml

```diff
@@ -0,0 +1,212 @@
+  | Generic_BSD, _ -> Generic_BSD
+  | Solaris, _ -> Solaris
+  | Dragonfly, _ -> Dragonfly
+  | GNU, _ -> GNU
+  | BeOS, _ -> BeOS
+  | Unknown, _ -> Unknown
+
+type flags_for_section = {
+  names : string list;
+  flags : string option;
+  args : string list;
+}
+
+let flags t ~first_occurrence =
+  let text () = [".text"], None, [] in
+  let data () = [".data"], None, [] in
+  let rodata () = [".rodata"], None, [] in
+  let system = derived_system () in
+  let names, flags, args =
+    match t, Target_system.architecture (), system with
+    | Text, _, _ -> text ()
+    | Data, _, _ -> data ()
+    | DWARF dwarf, _, MacOS_like ->
+      let name =
+        match dwarf with
+        | Debug_info -> "__debug_info"
+        | Debug_abbrev -> "__debug_abbrev"
+        | Debug_aranges -> "__debug_aranges"
+        | Debug_loc -> "__debug_loc"
+        | Debug_str -> "__debug_str"
+        | Debug_line -> "__debug_line"
+      in
+      ["__DWARF"; name], None, ["regular"; "debug"]
+    | DWARF dwarf, _, _ ->
+      let name =
+        match dwarf with
+        | Debug_info -> ".debug_info"
+        | Debug_abbrev -> ".debug_abbrev"
+        | Debug_aranges -> ".debug_aranges"
+        | Debug_loc -> ".debug_loc"
+        | Debug_str -> ".debug_str"
+        | Debug_line -> ".debug_line"
+      in
+      let flags =
+        if first_occurrence then
+          Some ""
+        else
+          None
+      in
+      let args =
+        if first_occurrence then
+          ["%progbits"]
+        else
+          []
+      in
+      [name], flags, args
+    | (Eight_byte_literals | Sixteen_byte_literals), (ARM | AArch64 | Z), _
+    | (Eight_byte_literals | Sixteen_byte_literals), _, Solaris ->
+      rodata ()
+    | Sixteen_byte_literals, _, MacOS_like ->
+      ["__TEXT";"__literal16"], None, ["16byte_literals"]
+    | Sixteen_byte_literals, _, (MinGW_64 | Cygwin) ->
+      [".rdata"], Some "dr", []
+    | Sixteen_byte_literals, _, (MinGW_32 | Win32 | Win64) ->
+      data ()
+    | Sixteen_byte_literals, _, _ ->
+      [".rodata.cst8"], Some "a", ["@progbits"]
+    | Eight_byte_literals, _, MacOS_like ->
+      ["__TEXT";"__literal8"], None, ["8byte_literals"]
+    | Eight_byte_literals, _, (MinGW_64 | Cygwin) ->
+      [".rdata"], Some "dr", []
+    | Eight_byte_literals, _, (MinGW_32 | Win32 | Win64) ->
+      data ()
+    | Eight_byte_literals, _, _ ->
+      [".rodata.cst8"], Some "a", ["@progbits"]
+    | Jump_tables, _, (MinGW_64 | Cygwin) ->
+      [".rdata"], Some "dr", []
+    | Jump_tables, _, (MinGW_32 | Win32) ->
+      data ()
+    | Jump_tables, _, (MacOS_like | Win64) ->
+      text () (* with LLVM/OS X and MASM, use the text segment *)
+    | Jump_tables, _, _ ->
+      [".rodata"], None, []
+    | Read_only_data, _, (MinGW_32 | Win32) ->
+      data ()
+    | Read_only_data, _, (MinGW_64 | Cygwin) ->
+      [".rdata"], Some "dr", []
+    | Read_only_data, _, _ ->
+      rodata ()
+  in
+  { names; flags; args; }
+
+let to_string t =
```

Comment:
This function is unused and not exported.

Reply:
This is now exported on the new asm_symbols branch